### ISSUE TITLE
Issue with Loss Computing

### ISSUE THREAD

**Issue by Pangolin112** [iteratetograceness/dreambooth-flux-pixel-icon#1]
Why would in training_script.py, line 1760 ~ 1763: the loss is computed using "mse_loss(model_pred - noise + model_input)", instead of "mse_loss(model_pred - noise)"?
`# Compute regular loss.
                loss = torch.mean(
                    (weighting.float() * (model_pred.float() - target.float()) ** 2).reshape(target.shape[0], -1),
                    1,
                )`

**Comment by iteratetograceness** [comment]
Hi @Pangolin112! Firstly, this script is pulled from Huggingface's `diffusers` training script: https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/README_flux.md

Secondly, my understanding is that this is expected given flux isn't predicting noise but using "flow matching"
